nxdrive/metrics/metrics.py

```diff
@@ -0,0 +1,61 @@
+from logging import getLogger
+from queue import Empty, Queue
+from time import monotonic_ns
+from typing import TYPE_CHECKING
+
+from ..constants import TIMEOUT
+from ..engine.workers import Worker
+from ..exceptions import ThreadInterrupt
+from ..objects import Metrics
+from ..options import Options
+from ..qt.imports import pyqtSlot
+from .constants import SYNC_ACTION, SYNC_TIME
+
+log = getLogger(__name__)
+
+_ENDPOINT = "/site/api/v1/me"
+
+if TYPE_CHECKING:
+    from ..client.remote_client import Remote
+
+
+class CustomMetrics(Worker):
+    def __init__(self, remote: "Remote") -> None:
+        super().__init__("CustomMetrics")
+
+        self._remote = remote
+        self._metrics_queue: Queue = Queue()
+        self._stop = False
+
+        self.timeout = Options.timeout if Options.timeout > 0 else TIMEOUT
+        self.thread.started.connect(self.run)
+
+    def _execute(self) -> None:
+        try:
+            while True:
+                try:
+                    metrics = self._metrics_queue.get(timeout=1)
+                    print(f"Sending {metrics}")  # DEV
+                    if not metrics or not Options.custom_metrics:
+                        continue
+                    self._remote.client.request("GET", _ENDPOINT, headers=metrics)
+                except Empty:
+                    continue
+                except Exception:
```

Comment:
Maybe should we handle connection errors and reput the metrics in the queue. Same for credentials issues.
In fact, we should be more smart here, because we would lose a lot of metrics then.